infrastructure/containerd/image_service_integration_test.go

```diff
@@ -0,0 +1,281 @@
+	Expect(len(img)).To(Equal(1))
+	Expect(img[0].Name).To(Equal(getTestVolumeImage()))
+
+	//
+	// Chapter II: Snapshots
+	//
+
+	snapshotExists := false
+	err = client.SnapshotService(testSnapshotter).Walk(namespaceCtx, func(walkCtx context.Context, info snapshots.Info) error {
+		if info.Name == expectedSnapshotName {
+			snapshotExists = true
+		}
+
+		return nil
+	})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(snapshotExists).To(BeTrue(), "expect snapshot with name %s to exist", expectedSnapshotName)
+
+	leases, err := client.LeasesService().List(namespaceCtx)
+	Expect(len(leases)).To(Equal(1))
+	Expect(leases[0].ID).To(Equal(expectedLeaseName), "expect lease with name %s to exists", expectedLeaseName)
+
+	//
+	// Chapter III: Kernel
+	//
+
+	inputGet.ImageName = getTestKernelImage()
+
+	err = imageSvc.Pull(ctx, inputGet)
+	Expect(err).NotTo(HaveOccurred())
+
+	exists, err := imageSvc.Exists(ctx, &ports.ImageSpec{
+		ImageName: getTestVolumeImage(),
+		Owner:     testOwnerUsageID,
+	})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(exists).To(BeTrue())
+
+	mounts, err = imageSvc.PullAndMount(ctx, &ports.ImageMountSpec{
+		ImageName:    getTestKernelImage(),
+		Owner:        testOwnerUsageID,
+		Use:          models.ImageUseKernel,
+		OwnerUsageID: testOwnerUsageID,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(mounts).NotTo(BeNil())
+	Expect(len(mounts)).To(Equal(1))
+}
+
+func testCreateClient(t *testing.T) (*ctr.Client, context.Context) {
+	addr := os.Getenv("CTR_SOCK_PATH")
+	client, err := ctr.New(addr)
+	Expect(err).NotTo(HaveOccurred())
+
+	ctx := context.Background()
+
+	serving, err := client.IsServing(ctx)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(serving).To(BeTrue())
+
+	return client, ctx
+}
+
+func runContainerDTests() bool {
+	testCtr := os.Getenv("CTR_SOCK_PATH")
+	return testCtr != ""
+}
+
+// getTestVolumeImage returns with the default test volume image name, if
+// CTR_TEST_VOL_IMG environment variable is not defined.
+func getTestVolumeImage() string {
+	envImage := os.Getenv("CTR_TEST_VOL_IMG")
+	if envImage != "" {
+		return envImage
+	}
+
+	return testImageVolume
+}
+
+// getTestKernelImage returns with the default test kernel image name, if
+// CTR_TEST_KERNEL_IMG environment variable is not defined.
+func getTestKernelImage() string {
+	envImage := os.Getenv("CTR_TEST_KERNEL_IMG")
+	if envImage != "" {
+		return envImage
+	}
+
+	return testImageKernel
+}
+
+type testImageMountOptions struct {
+	ImageName string
+	Owner     string
+	Use       models.ImageUse
+	Expected  bool
+}
+
+// testImageMounted checks that images have been mounted
+// tests the IsMounted method
+func testImageMounted(ctx context.Context, imageSvc ports.ImageService, opts testImageMountOptions) {
+	mounted, err := imageSvc.IsMounted(ctx, &ports.ImageMountSpec{
+		ImageName:    opts.ImageName,
+		Owner:        opts.Owner,
+		Use:          opts.Use,
+		OwnerUsageID: testOwnerUsageID,
+	})
+	Expect(err).ToNot(HaveOccurred())
```

Comment:
```suggestion
	Expect(err).NotTo(HaveOccurred())
```
:trollface: :trollface:

Reply:
But but but but bu but... `to not have occurred` vs `not to have occurred`, the first one sounds better to my ears.

Reply:
honestly they both sound off to me, i have had this discussion like 100 times 😂  i have a vim shortcut which autocompletes to NotTo and I have no idea why

everywhere else we have `NotTo` so for consistency i guess?

Reply:
```
flintlock on  15-review-oci-tests via  v1.17.1
❯ ag 'NotTo' | wc -l
97

flintlock on  15-review-oci-tests via  v1.17.1
❯ ag 'ToNot' | wc -l
31

flintlock on  15-review-oci-tests via  v1.17.1
❯ ag -l 'NotTo'
core/models/vmid_test.go
core/application/app_test.go
core/steps/runtime/dir_create_test.go
core/plans/microvm_delete_test.go
core/plans/microvm_create_update_test.go
test/e2e/utils/utils.go
test/e2e/utils/runner.go
pkg/wait/wait_test.go
pkg/log/log_test.go
pkg/validation/validate_test.go
pkg/planner/actuator_test.go
infrastructure/containerd/repo_test.go
infrastructure/containerd/image_service_integration_test.go
infrastructure/controllers/microvm_controller_test.go
infrastructure/network/utils_test.go

flintlock on  15-review-oci-tests via  v1.17.1
❯ ag -l 'ToNot'
core/models/vmid_test.go
core/steps/runtime/kernel_mount_test.go
core/steps/runtime/initrd_mount_test.go
core/steps/runtime/repo_release_test.go
core/steps/runtime/volume_mount_test.go
core/steps/microvm/start_test.go
core/steps/microvm/create_test.go
core/steps/microvm/delete_test.go
core/steps/network/interface_create_test.go
core/steps/network/interface_delete_test.go
core/steps/event/publish_test.go
test/e2e/utils/utils.go
test/e2e/e2e_test.go
infrastructure/containerd/image_service_test.go
infrastructure/containerd/image_service_integration_test.go
```

it's more like 2/3.

Reply:
lool and some of those ToNots were mine 🤦‍♀️